lib/ansible/modules/cloud/online/online_server_facts.py

```diff
@@ -0,0 +1,176 @@
+                    "$ref": "/api/v1/server/hardware/disk/68453"
+                }
+            ],
+            "drive_arrays": [
+                {
+                    "disks": [
+                        {
+                            "$ref": "/api/v1/server/hardware/disk/68452"
+                        },
+                        {
+                            "$ref": "/api/v1/server/hardware/disk/68453"
+                        }
+                    ],
+                    "raid_controller": {
+                        "$ref": "/api/v1/server/hardware/raidController/9910"
+                    },
+                    "raid_level": "RAID1"
+                }
+            ],
+            "hardware_watch": true,
+            "hostname": "sd-42",
+            "id": 42,
+            "ip": [
+                {
+                    "address": "195.154.172.149",
+                    "mac": "28:92:4a:33:5e:c6",
+                    "reverse": "195-154-172-149.rev.poneytelecom.eu.",
+                    "switch_port_state": "up",
+                    "type": "public"
+                },
+                {
+                    "address": "10.90.53.212",
+                    "mac": "28:92:4a:33:5e:c7",
+                    "reverse": null,
+                    "switch_port_state": "up",
+                    "type": "private"
+                }
+            ],
+            "last_reboot": "2018-08-23T08:32:03.000Z",
+            "location": {
+                "block": "A",
+                "datacenter": "DC3",
+                "position": 19,
+                "rack": "A23",
+                "room": "4 4-4"
+            },
+            "network": {
+                "ip": [
+                    "195.154.172.149"
+                ],
+                "ipfo": [],
+                "private": [
+                    "10.90.53.212"
+                ]
+            },
+            "offer": "Pro-1-S-SATA",
+            "os": {
+                "name": "FreeBSD",
+                "version": "11.1-RELEASE"
+            },
+            "power": "ON",
+            "proactive_monitoring": false,
+            "raid_controllers": [
+                {
+                    "$ref": "/api/v1/server/hardware/raidController/9910"
+                }
+            ],
+            "support": "Basic service level"
+        }
+    ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urljoin
+from ansible.module_utils.online import (
+    Online, OnlineException, online_argument_spec
+)
+
+
+class OnlineServerFacts(Online):
+
+    def __init__(self, module):
+        super(OnlineServerFacts, self).__init__(module)
+        self.name = 'api/v1/server'
+
+    def _get_server_detail(self, server_path):
+        try:
+            return self.get(path=server_path).json
+        except OnlineException:
+            self.module.warning("A problem occured while fetching: %s" % server_path)
+            raise OnlineException
```

Comment:
`self.module.fail_json` should be called right here (`fail_json` is called within an `except` block, no need to use `exception` parameter):
```
except OnlineException as exc:
    self.module.fail_json(msg="A problem occurred while fetching: %s (%s)" % (server_path, exc))
```